src/main/java/no/unit/nva/amazon/s3/CreateUploadHandler.java

```diff
@@ -1,162 +1,115 @@
-
-            String keyName = UUID.randomUUID().toString();
-            InitiateMultipartUploadRequest initRequest =
-                    new InitiateMultipartUploadRequest(bucketName, keyName, objectMetadata);
-            InitiateMultipartUploadResult initResponse = getS3Client().initiateMultipartUpload(initRequest);
-            System.out.println(initResponse);
-            CreateUploadResponseBody responseBody = new CreateUploadResponseBody(initResponse.getUploadId(), keyName);
-            response.setBody(new Gson().toJson(responseBody));
-            response.setStatusCode(SC_CREATED);
-        } catch (SdkClientException e) {
-            System.out.println(DebugUtils.dumpException(e));
-            response.setErrorBody(e.getMessage());
-            response.setStatusCode(SC_INTERNAL_SERVER_ERROR);
-        } catch (Exception e) {
-            System.out.println(DebugUtils.dumpException(e));
-            response.setErrorBody(e.getMessage());
-            response.setStatusCode(SC_INTERNAL_SERVER_ERROR);
-            return response;
-        }
-        return response;
-    }
-
     /**
      * Extracting metadata from the given file resource.
      * @param requestBody incoming parameters
      * @return Metadata to be stored with file on S3
      */
-    public ObjectMetadata getObjectMetadata(CreateUploadRequestBody requestBody) {
+    public ObjectMetadata toObjectMetadata(CreateUploadRequestBody requestBody) {
         ObjectMetadata objectMetadata = new ObjectMetadata();
         objectMetadata.setContentMD5(null);
+        objectMetadata.setContentDisposition(toContentDisposition(requestBody.getFilename()));
+        objectMetadata.setContentType(requestBody.getMimetype());
+        return objectMetadata;
+    }
 
-        if (!Objects.isNull(requestBody.getFilename()) && !requestBody.getFilename().isEmpty()) {
-            objectMetadata.setContentDisposition("filename=\"" + requestBody.getFilename() + "\"");
-        }
-        if (!(Objects.isNull(requestBody.getMimetype()) || requestBody.getMimetype().isEmpty())
-                && requestBody.getMimetype().contains("/")) {
-            objectMetadata.setContentType(requestBody.getMimetype());
-        }
+    protected String toContentDisposition(String filename) {
+        return String.format("filename=\"%s\"", filename);
+    }
 
-        return objectMetadata;
+
+    @Override
+    protected CreateUploadResponseBody processInput(CreateUploadRequestBody input, RequestInfo requestInfo,
+                                                    Context context) throws ApiGatewayException {
+        validate(input);
+
+        String keyName = UUID.randomUUID().toString();
+        InitiateMultipartUploadRequest initRequest =
+                new InitiateMultipartUploadRequest(bucketName, keyName, toObjectMetadata(input));
+        InitiateMultipartUploadResult initResponse = s3Client.initiateMultipartUpload(initRequest);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code manually manages uploads or downloads. The SDK can automatically handle these operations. Consider using the SDK to automatically handle this instead. SDK class: TransferManager. See [Using TransferManager for Amazon S3 Operations](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/examples-s3-transfermanager.html).